get_activations.py

```diff
@@ -0,0 +1,135 @@
+
+    return parser.parse_args()
+
+
+
+class Trainer:
+    def __init__(self, args, device):
+        self.args = args
+        self.device = device
+        model = model_factory.get_network(args.network)(jigsaw_classes=args.jigsaw_n_classes + 1, classes=args.n_classes)
+        self.model = model.to(device)
+        # print(self.model)
+        #self.source_loader, self.val_loader = data_helper.get_train_dataloader(args, patches=model.is_patch_based())
+        self.target_loader = data_helper.get_val_dataloader(args, patches=model.is_patch_based())
+        #self.test_loaders = {"val": self.val_loader, "test": self.target_loader}
+        #self.len_dataloader = len(self.source_loader)
+        print("Dataset size: test %d" % (len(self.target_loader.dataset)))
+        self.optimizer, self.scheduler = get_optim_and_scheduler(model, args.epochs, args.learning_rate, args.train_all, nesterov=args.nesterov)
+        self.jig_weight = args.jig_weight
+        self.only_non_scrambled = args.classify_only_sane
+        self.n_classes = args.n_classes
+        if args.target in args.source:
+            self.target_id = args.source.index(args.target)
+            print("Target in source: %d" % self.target_id)
+            print(args.source)
+        else:
+            self.target_id = None
+
+        self.best_val_jigsaw = 0.0
+        self.best_jigsaw_acc = 0.0
+        folder_name, logname = Logger.get_name_from_args(args)
+        
+
+        self.folder_name = folder_name
+        #self.save_folder = os.path.join("logs", folder_name, logname)
+
+    def hook(self, model, input, output):
+        output = output.view(output.size(0), -1)
+        print(output.size())
+        self.activations.append(output.detach().cpu().numpy())
+
+    def get_features(self):
+        loader = self.target_loader
+        model_path = osp.join('logs',self.folder_name,'best_model.pth')
+        self.activations = []
+        sd = torch.load(model_path)
+        self.model.load_state_dict(sd['model_state_dict'])
+        self.model.avgpool.register_forward_hook(self.hook)
+        self.model.eval()
+        pbar = pkbar.Pbar(name='Epoch Progress',target=len(loader))
+        labels = []
+        for it, ((data, jig_l, class_l), _) in enumerate(loader):
+            pbar.update(it)
+            data, jig_l, class_l = data.to(self.device), jig_l.to(self.device), class_l.to(self.device)
+            jigsaw_logit, class_logit = self.model(data)
+            labels.append(class_l.detach().cpu().numpy())
+        self.activations = functools.reduce((lambda x,y: np.append(x,y,axis=0)),self.activations)
+        labels = functools.reduce((lambda x,y: np.append(x,y,axis=0)),labels)
+        
+        target_data = {}
+        target_data['features'] = self.activations
+        target_data['labels'] = labels
+        with open(osp.join('logs',self.folder_name, 'act_labels.pkl'), 'wb') as handle:
```

Comment:
self.folder_name doesn't save it inside the current experiment folder, it saves it in it's parent.
I'm changing it to self.save_folder.